hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/validator/SparkPreCommitValidator.java

```diff
@@ -73,7 +79,23 @@ public void validate(String instantTime, HoodieWriteMetadata<O> writeResult, Dat
     try {
       validateRecordsBeforeAndAfter(before, after, getPartitionsModified(writeResult));
     } finally {
-      LOG.info(getClass() + " validator took " + timer.endTimer() + " ms");
+      long duration = timer.endTimer();
+      LOG.info(getClass() + " validator took " + duration + " ms" + ", metrics on? " + getWriteConfig().isMetricsOn());
+      publishRunStats(instantTime, duration);
+    }
+  }
+
+  /**
+   * Publish pre-commit validator run stats for a given commit action.
+   */
+  protected void publishRunStats(String instantTime, long duration) {
+    // Record validator duration metrics.
```

Comment:
protected -> private

Reply:
Thanks for catching this, updated the scope to private.